sample_factory/algorithms/appo/learner.py

```diff
@@ -397,8 +397,12 @@ def _calculate_gae(self, buffer):
         This is leftover the from previous version of the algorithm.
         Perhaps should be re-implemented in PyTorch tensors, similar to V-trace for uniformity.
         """
+        ent_coeff = self.cfg.ppo_max_entropy_coeff
```

Comment:
I agree with Alex's main points, max entropy is definitely also compatible (and probably useful) with V-trace.

In garage we _could_ differentiate back to the entropy through the returns, since our return calculation is differentiable, but we made doing so [an explicit error](https://github.com/rlworkgroup/garage/blob/master/src/garage/tf/algos/npo.py#L553).

In other words, they should not be differentiable. BTW, you might want to leave a comment pointing to [this paper](https://arxiv.org/abs/1805.00909) somewhere.

Reply:
@krzentner What part of the paper is relevant here? I will add that contextual info in as well

Reply:
@alex-petrenko yes we can add it before calling `_calculate_gae(buffer)`. But notice that I am only changing the rewards locally, not changing `buffer.rewards`. 

Do you want me to update the buffer with new rewards (it might be reserved only for task rewards, independent of entropy. I'm not sure)? If not, we can pass in `rewards+buffer.weighted_entropies` (new variable inside `buffer`) when we calculat GAE. And tell me what I should change for vtrace.

Reply:
I think it is okay to just modify the rewards directly. Then you don't need to do anything special for v-trace - it will automatically use the new reward values.